src/dataframe/dataframe.jl

```diff
@@ -303,6 +303,8 @@ function Base.getindex(df::DataFrame, col_inds::AbstractVector)
     return DataFrame(new_columns, Index(_names(df)[selected_columns]))
 end
 
+Base.getindex(df::AbstractDataFrame, col_inds::Regex) = getindex(df, index(df)[col_inds])
```

Comment:
Why not merge this with previous method? Same below.

Reply:
This would lead to method ambiguities.